banzai_nres/utils/trace_utils.py

```diff
@@ -0,0 +1,538 @@
+    :param image_width: image.data.shape[1]
+    :return: trace centroids for each trace, versus x pixel. E.g. trace_values_versus_xpixel[2,5] is the 3rd orders value
+            at x=5.
+            num_traces = num_orders*Num_fibers
+            x = [0,1,2,...,image.data.shape[1]-1]
+    """
+    coeflen, coefwidth = coefficients_and_indices.shape
+    num_traces, order_of_poly_fits = coeflen, coefwidth - 2
+    legendre_polynomial_array, x, xnorm = generate_legendre_array(image_width, order_of_poly_fits)
+    trace_values_versus_xpixel = np.dot(coefficients_and_indices[:, 1:], legendre_polynomial_array)
+    return trace_values_versus_xpixel, num_traces, x
+
+
+def convert_numpy_array_coefficients_to_astropy_table(coefficients_table_name='',
+                                                      fiber_order=None, coefficients=None):
+    order_numbers = list(coefficients[:, 0].astype(np.int))
+    coefficients_table = generate_astropy_table_from_numpy_array_and_orders(coefficients_table_name,
+                                                                            order_numbers,
+                                                                            coefficients[:, 1:],
+                                                                            fiber_order)
+    coefficients_table[coefficients_table_name].description = 'Legendre polynomials ' \
+                                                               'normalized from -1 to 1 over the ' \
+                                                               'number of columns of the image, ' \
+                                                               'usually 0 to 4095'
+    return coefficients_table
+
+
+def convert_numpy_array_trace_centroids_to_astropy_table(trace_centroids, coefficients,
+                                                         trace_center_table_name='', fiber_order=None):
+    order_numbers = list(coefficients[:, 0].astype(np.int))
+    trace_centroids_table = generate_astropy_table_from_numpy_array_and_orders(trace_center_table_name,
+                                                                               order_numbers,
+                                                                               trace_centroids,
+                                                                               fiber_order)
+    trace_centroids_table[trace_center_table_name].unit = 'pixel'
+    trace_centroids_table[trace_center_table_name].description = 'y pixel position for trace' \
+                                                                                     'center at each x pixel' \
+                                                                                     'from 0 to number of columns' \
+                                                                                     'in the image'
+    return trace_centroids_table
+
+
+def convert_astropy_table_coefficients_to_numpy_array(astropy_table_of_coefficients, coefficients_table_name=''):
```

Comment:
Why do we need to convert between the 2? The data in an astropy table is a numpy array.